compat/mingw.c

```diff
@@ -2381,7 +2382,37 @@ int link(const char *oldpath, const char *newpath)
 	return 0;
 }
 
-int symlink(const char *target, const char *link)
+enum symlink_type {
+	SYMLINK_TYPE_UNSPECIFIED = 0,
+	SYMLINK_TYPE_FILE,
+	SYMLINK_TYPE_DIRECTORY,
+};
+
+static enum symlink_type check_symlink_attr(struct index_state *index, const char *link)
+{
+	static struct attr_check *check;
+	const char *value;
+
+	if (!index)
+		return SYMLINK_TYPE_UNSPECIFIED;
+
+	if (!check)
+		check = attr_check_initl("symlink", NULL);
+
+	git_check_attr(index, link, check);
+
+	value = check->items[0].value;
+	if (value == NULL)
+		;
+	else if (!strcmp(value, "file"))
+		return SYMLINK_TYPE_FILE;
+	else if (!strcmp(value, "dir"))
+		return SYMLINK_TYPE_DIRECTORY;
+
+	return SYMLINK_TYPE_UNSPECIFIED;
```

Comment:
Maybe issue a warning if the attribute is set to an unknown value.

Reply:
Yep, and I also handle "directory" now, for good measure.